test/controllers/test_party_controller.py

```diff
@@ -0,0 +1,133 @@
+    def test_get_enrolment_data(self, mock_engine):
+        row_1, row_2, row_3 = self._generate_3_enrolment_data_rows()
+        mock_engine.engine.execute().all.return_value = [row_1, row_2, row_3]
+        expected_output = [row_1, row_2, row_3]
+        business_ids = f"{business_1_id}, {business_2_id}"
+        test_output = party_controller.get_enrolment_data(survey_id, business_ids)
+        self.assertEqual(expected_output, test_output)
+
+    def test_get_enrolment_data_empty_string(self):
+        test_business_ids = ""
+        expected_output = []
+        test_output = party_controller.get_enrolment_data(survey_id, test_business_ids)
+        self.assertEqual(expected_output, test_output)
+
+    def test_get_respondent_ids_from_enrolment_data(self):
+        row_1, row_2, row_3 = self._generate_3_enrolment_data_rows()
+        test_input = [row_1, row_2, row_3]
+        expected_output = f"'{respondent_1_id}', '{respondent_2_id}', '{respondent_3_id}'"
+        test_output = party_controller.get_respondent_ids_from_enrolment_data(test_input)
+        self.assertEqual(expected_output, test_output)
+
+    def test_get_respondent_ids_from_enrolment_data_empty_string(self):
+        test_input = []
+        expected_output = ""
+        test_output = party_controller.get_respondent_ids_from_enrolment_data(test_input)
+        self.assertEqual(expected_output, test_output)
+
+    @mock.patch("rm_reporting.app.party_db")
+    def test_get_respondent_data(self, mock_engine):
+        row_1, row_2 = self._generate_2_respondent_data_rows()
+        mock_engine.engine.execute().all.return_value = [row_1, row_2]
+        expected_output = {"1": row_1, "2": row_2}
+        respondent_ids = f"{respondent_1_id}, {respondent_2_id}"
+        test_output = party_controller.get_respondent_data(respondent_ids)
+        self.assertEqual(expected_output, test_output)
+
+    def test_get_respondent_data_empty_string(self):
+        test_input = ""
+        expected_output = {}
+        test_output = party_controller.get_respondent_data(test_input)
+        self.assertEqual(expected_output, test_output)
+
+    @staticmethod
+    def _generate_2_respondent_data_rows() -> tuple[Row, Row]:
+        row_1 = Row(
+            id="1",
+            party_uuid=respondent_2_id,
+            status="ACTIVE",
+            email_address="jackjohnson@email.com",
+            pending_email_address="",
+            first_name="Jack",
+            last_name="Johnson",
+            telephone="01234123123",
+            mark_for_deletion=False,
+            created_on="2022-12-04 15:22:20.815615",
+            password_verification_token="",
+            password_reset_counter=0,
+        )
+        row_2 = Row(
+            id="2",
+            party_uuid=respondent_2_id,
+            status="ACTIVE",
+            email_address="petergriffin@email.com",
+            pending_email_address="",
+            first_name="Peter",
+            last_name="Griffin",
```

Comment:
Family survey?

Reply:
Just needed a name and it was the first to come to mind ^^